api/v1alpha3/conversion.go

```diff
@@ -329,6 +331,16 @@ func restoreInstance(restored, dst *v1alpha4.Instance) {
 	restoreNonRootVolumes(restored.NonRootVolumes, dst.NonRootVolumes)
 }
 
+// Manually restore the control plane loadbalancer data
+func restoreControlPlaneLoadBalancer(restored, dst *v1alpha4.AWSLoadBalancerSpec) {
+	if restored == nil {
+		return
+	}
+	if dst.Name == nil {
```

Comment:
The change looks great. 

I think `if dst.Name == nil` will be always true in this context.

Reply:
Thanks, fixed!

While looking at this, I spotted some problems in other restore functions: #2809. Would appreciate your feedback there.